modules/tensor_mechanics/src/materials/ComputeBlockRotatedElasticityTensor.C

```diff
@@ -24,26 +25,28 @@ validParams<ComputeBlockRotatedElasticityTensor>()
       "fill_method", RankFourTensor::fillMethodEnum() = "symmetric9", "The fill method");
   params.addRequiredParam<UserObjectName>("euler_angle_provider",
                                           "Name of Euler angle provider user object");
+  params.addRequiredParam<int>("offset", "Offset value for grain_id");
+
   return params;
 }
 
-ComputeBlockRotatedElasticityTensor::ComputeBlockRotatedElasticityTensor(const InputParameters & parameters)
+ComputeBlockRotatedElasticityTensor::ComputeBlockRotatedElasticityTensor(
+    const InputParameters & parameters)
   : ComputeElasticityTensorBase(parameters),
-
     _C_ijkl(getParam<std::vector<Real>>("C_ijkl"),
             (RankFourTensor::FillMethod)(int)getParam<MooseEnum>("fill_method")),
-    _euler(getUserObject<EulerAngleProvider>("euler_angle_provider"))
+    _euler(getUserObject<EulerAngleProvider>("euler_angle_provider")),
+    _offset(getParam<int>("offset"))
 {
 }
-
 void
 ComputeBlockRotatedElasticityTensor::computeQpElasticityTensor()
 {
   EulerAngles angles;
   auto grain_id = _current_elem->subdomain_id();
-  if (grain_id == 0)
-    mooseError("ZERO!!!!");
-  grain_id--;
+  if (grain_id == 0 && _offset != 0)
```

Comment:
no. you should subtract the offset first and then error out if the resulting value is < 0 or >= num grains.